### PR TITLE
In-memory engine: consider tikv gc in cross check

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref #16141

What's Changed:

```commit-message
consider tikv gc in cross check
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/server/src/common.rs [hunk 1]
```diff
@@ -715,7 +717,14 @@ pub fn build_hybrid_engine(
         )
     }
 
-    memory_engine.start_cross_check(disk_engine.clone(), region_cache_engine_context.pd_client());
+    memory_engine.start_cross_check(
+        disk_engine.clone(),
+        region_cache_engine_context.pd_client(),
+        Box::new(|| {
+            let ctx = GC_CONTEXT.lock().unwrap();
```
Comment (glorv) [inline components/server/src/common.rs:724]: What if compaction_filter is not enable? Seems this `GC_CONTEXT` can only be used when compaction_filter is enabled.

**Review by glorv — COMMENTED** [review]
rest LGTM

File: components/server/src/common.rs (hunk 1 above)
Comment (SpadeA-Tang) [reply components/server/src/common.rs:724]: Good catch. For simplicity, cross check is only supported when compaction filter is enabled.

File: components/in_memory_engine/src/cross_check.rs [hunk 2]
```diff
@@ -838,9 +881,15 @@ impl Runnable for CrossChecker {
 
         let now = Instant::now();
 
-        ranges_to_audit
-            .into_iter()
-            .for_each(|r| self.cross_check_range(&r, &snap));
+        ranges_to_audit.into_iter().for_each(|r| {
+            if let Err(e) = self.cross_check_region(&r, &snap) {
+                info!(
+                    "cross check stopped";
```
Comment (overvenus) [inline components/in_memory_engine/src/cross_check.rs:887]: Please add "ime" prefix for logs in this crate.

**Review by overvenus — COMMENTED** [review]
Rest LGTM

**Review by overvenus — APPROVED** [review]

**Review by glorv — APPROVED** [review]

**Comment by SpadeA-Tang** [comment]
PTAL thansk @zhangjinpeng87

**Review by zhangjinpeng87 — APPROVED** [review]
LGTM

**Pull request merged**